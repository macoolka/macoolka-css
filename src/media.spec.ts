
import { OutRule,OutProps, parseRule,theme } from '.';

/**
 * Enum Properties's value using string or number,
 * The tag 'ruleEnum' in rule describe the property style
 */
export type EProp = {
    size?: 'small' | 'medium' 
};
/**
 * Stand Properties's value using any type,
 * The tag 'rule' in rule describe the property style
 */
export type SProp = {
    disabled?: boolean
};
/**
 * The output props. including predefine property and orginal property
 */
export type Props = OutProps<EProp & SProp>;

export const rule: OutRule<SProp, EProp> = {
    /**
     * default style
     */
    style: {
        mkFontSize:'h3',
        selector: {
            ':focus': {
               width:30,
            },
        },
        mkMedia:[{width:1,selector:{':hover':{width:10}}},{width:2,selector:{':hover':{width:20}}},{width:3,selector:{':hover':{width:30}}}]
    },
    /**
     * descirbe stand property.
     * the type is
     * propertyName :({value,name,theme,source})=>cssnode
     * value is current value.
     * name is propertyname
     * theme is gloabe theme
     * source is component props
     */
    rule: {
        disabled: ({value})=> ({
            opacity: value ? 0.38 : 1,
        }),
    },
    /**
     * descirbe enum property.
     * the type is
     * propertyName :{propertyvalue:({value,name,theme,source})=>cssnode}
     */
    ruleEnum: {
        size: {
            small: ()=>({
                mkFontSize:'h1',
                mkHoverTextColor:'accent',
                selector: {
                    ':focus': {
                       width:10,
                    },
                },
                mkMedia:[{width:11,selector:{':hover':{width:110}}},{width:2,selector:{':hover':{width:120}}},{width:13,selector:{':hover':{width:130}}}]
            }),
            medium:()=>({
                mkFontSize:'h2',
                mkMedia:[{mkPadding:'small'},{mkPadding:'medium'}],
                selector: {
                    ':focus': {
                       width:10,
                    },
                },
            }),

        },
    },
};
export const parse = parseRule(rule,theme);
describe('should parse component rule', () => {
    it('1 should parse empty props', () => {
        expect(parse({
        })).toMatchSnapshot()
    });
    it('2 should parse common props', () => {
        expect(parse({
            size:'small',
        })).toMatchSnapshot()
    });
 
});