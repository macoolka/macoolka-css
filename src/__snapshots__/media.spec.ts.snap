// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse component rule 1 should parse empty props 1`] = `
"font-size: 48px;
:focus {
  width: 30px;
}
@media screen and (max-width: 93em) {
  width: 3px;
}
@media screen and (max-width: 80em) {
  width: 2px;
}
@media screen and (max-width: 50em) {
  width: 100%;
}
@media screen and (max-width: 93em) {
  :hover {
    width: 30px;
  }
}
@media screen and (max-width: 80em) {
  :hover {
    width: 20px;
  }
}
@media screen and (max-width: 50em) {
  :hover {
    width: 10px;
  }
}"
`;

exports[`should parse component rule 2 should parse common props 1`] = `
"font-size: 96px;
&:hover {
  color: #f50057;
}
:focus {
  width: 10px;
}
@media screen and (max-width: 93em) {
  width: 13px;
}
@media screen and (max-width: 80em) {
  width: 2px;
}
@media screen and (max-width: 50em) {
  width: 11px;
}
@media screen and (max-width: 93em) {
  :hover {
    width: 130px;
  }
}
@media screen and (max-width: 80em) {
  :hover {
    width: 120px;
  }
}
@media screen and (max-width: 50em) {
  :hover {
    width: 110px;
  }
}"
`;
